pyfar/signal.py

```diff
@@ -444,6 +193,416 @@ def __setitem__(self, key, value):
+            is 'none', which is typically used for energy signals, such as
+            impulse responses.
+        comment : str
+            A comment related to the data. The default is None.
+        dtype : string, optional
+            Raw data type of the signal. The default is float64
+
+        References
+        ----------
+        .. [1] J. Ahrens, C. Andersson, P. Höstmad, and W. Kropp, “Tutorial on
+               Scaling of the Discrete Fourier Transform and the Implied
+               Physical Units of the Spectra of Time-Discrete Signals,” Vienna,
+               Austria, May 2020, p. e-Brief 600.
+
+        """
+
+        Audio.__init__(self, 'freq', comment, dtype)
+
+        # init data
+        self._data = np.atleast_2d(np.asarray(data, dtype=np.complex))
+
+        # init frequencies
+        self._frequencies = np.atleast_1d(np.asarray(frequencies).flatten())
+        if self._frequencies.size != self.n_bins:
+            raise ValueError(
+                "The number of freqencies must be data.shape[-1]")
+
+        if fft_norm is None:
+            fft_norm = 'none'
+        if fft_norm in self._VALID_FFT_NORMS:
+            self._fft_norm = fft_norm
+        else:
+            raise ValueError(("Invalid FFT normalization. Has to be "
+                              f"{', '.join(self._VALID_FFT_NORMS)}, but found "
+                              f"'{fft_norm}'"))
+
+    @property
+    def freq(self):
+        """The data in the frequency domain."""
+        if isinstance(self, Signal):
+            # the setter also converts the data from 'time' to 'freq'
+            self.domain = 'freq'
+        else:
+            # this only changes the class internal variable
+            self._domain = 'freq'
+        return self._data
+
+    @property
+    def frequencies(self):
+        """Frequencies of the discrete signal spectrum."""
+        return self._frequencies
+
+    @property
+    def n_bins(self):
+        """Number of frequency bins."""
+        return self._data.shape[-1]
+
+    @freq.setter
+    def freq(self, value):
+        spec = np.atleast_2d(np.atleast_2d(value))
+        new_num_bins = spec.shape[-1]
+        self._data = spec
+        self._domain = 'freq'
+
+        # also set n_samples in case of a Signal object
+        if isinstance(self, Signal):
+            if new_num_bins == self.n_bins:
+                n_samples = self.n_samples
+            else:
+                warnings.warn(UserWarning((
+                    "Number of frequency bins changed, assuming an even "
+                    "number of samples from the number of frequency bins.")))
+                n_samples = (new_num_bins - 1)*2
+            self._n_samples = n_samples
+
+    @property
+    def fft_norm(self):
+        """
+        The normalization for the Fourier Transform.
+
+        See pyfar.fft.normalization for more information.
+        """
+        return self._fft_norm
```

Comment:
Yes - the values of the frequency data may change with the FFT norm (e.g. a factor of two for all frequencies except 0 Hz and Nyquist). This will be important, if we later have a function that interpolates `FrequencyData` to a complete half-sided spectrum that can be entered into a `Signal`